lib/vecmat.gi

```diff
@@ -1411,12 +1411,12 @@ local sf, rep, ind, ind2, row, i,big,l;
   else
     if not IsField(field) then
       # not a field
-      return Immutable(matrix);
+      return matrix;
     fi;
     sf:=Size(field);
   fi;
 
-  big:=sf>256 or sf=0;
+  big:=sf>256 or sf=0 or (sf <= 256 and not field=GF(sf));
```

Comment:
The `sf <= 256` is redundant, due to the checks before it.

I am also a bit wary of writing `field=GF(sf)`, and its performance impact. Of course if this is called a lot with the same value, it won't matter due to `GFCACHE`; but if `sf` alternates between different field sizes. But then again, I am perhaps worrying too much (sure, there is a primality check, and for small `sf` it's fast, and then we end up using `GALOIS_FIELDS`, the second cache for small galois fields...)

Anyway, how about using `IsFFECollection`, then this is simply avoided. I.e.:

`big:=sf>256 or sf=0 or not IsFFECollection(field);`